scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/Job.java

```diff
@@ -358,6 +379,21 @@ public static Job parseJSONInfo(JSONObject jsonJobInfo) {
                        finishedTime);
     }
 
+    private static Map<String, String> extractGenericInformation(JSONValue jsonGenericInformation) {
+        Map<String, String> genericInformation = new HashMap<>();
+        if (jsonGenericInformation != null) {
+            JSONArray genericInformationArray = jsonGenericInformation.isArray();
+            if (genericInformationArray != null) {
+                for (int i = 0; i < genericInformationArray.size(); i++) {
+                    JSONObject genericInformationObject = genericInformationArray.get(i).isObject();
+                    genericInformation.put(genericInformationObject.get("key").isString().stringValue(),
```

Comment:
The code could be factorized by extracting `genericInformationObject.get("key").isString().stringValue()` to a method that takes `"key"` as argument.